### ISSUE TITLE
redundant_type_annotation ignores arrays

### ISSUE THREAD

**Issue by theoriginalbit** [realm/SwiftLint#3141]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`redundant_type_annotation` ignores types that are arrays.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
Done linting! Found 0 violations, 0 serious in 47 files.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.39.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
brew
* Paste your configuration file:

```yml
disabled_rules:
  - line_length
  - function_parameter_count
  - large_tuple
opt_in_rules:
  - attributes
  - file_header
  - empty_string
  - closure_spacing
  - convenience_type
  - discouraged_object_literal
  - fatal_error_message
  - force_unwrapping
  - implicitly_unwrapped_optional
  - yoda_condition
  - trailing_closure
  - redundant_type_annotation
included:
  - Source
excluded:
  - "Source/R.generated.swift"

attributes:
    always_on_same_line: ["@IBAction", "@NSManaged", "@objc"]
file_header:
    forbidden_pattern: ".?"
identifier_name:
  excluded:
    - id
trailing_comma:
  mandatory_comma: true
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  Nope
* Which Xcode version are you using (check `xcodebuild -version`)?
Xcode 11.3.1
Build version 11C504
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
```bash
$ echo "let values: [Int] = [Int]()" | swiftlint lint --no-cache --use-stdin --enable-all-rules
<nopath>:1:1: warning: Explicit ACL Violation: All declarations should specify Access Control Level keywords explicitly. (explicit_acl)
<nopath>:1:1: warning: Explicit Top Level ACL Violation: Top-level declarations should specify Access Control Level keywords explicitly. (explicit_top_level_acl)
<nopath>:1:5: warning: Prefixed Top-Level Constant Violation: Top-level constants should be prefixed by `k`. (prefixed_toplevel_constant)
Done linting! Found 3 violations, 0 serious in 1 file.
```

```swift
let values: [Int] = [Int]()
```

**Comment by theoriginalbit** [comment]
Please don't close stale bot.

**Comment by theoriginalbit** [comment]
c'mon stale bot. be kind.

**Comment by theoriginalbit** [comment]
![](https://media1.tenor.com/images/9f1eba07b0e6948af625b370a742bf5f/tenor.gif?itemid=17565387)

**Comment by jpsim** [comment]
Haha, sorry I ended up changing my mind on stalebot (https://github.com/realm/SwiftLint/pull/3529) but thanks for taking this with some amount of levity 😄. Let's keep this open. This is a bug, I haven't looked into fixing but I suspect it'll require using more of a tokenization approach to detect types involved rather than what's likely a regex here.

**Comment by theoriginalbit** [comment]
yeah no worries. thanks for re-opening 😄

**Comment by UksusoFF** [comment]
Same trouble with 
```
let value: Int = 0
```
Seems it's can be fixed as well as #3438

**Comment by revolter** [comment]
Same for strings:

```swift
static var storyboardName: String = "MyStoryboard"
```
